File: leetcode-submissions/0643-maximum-average-subarray-i/solution.java
class Solution {
    public double findMaxAverage(int[] nums, int k) {
        //sliding window, subarray size k, average
        //
        double currAverage = 0;
        double currSum = 0;
        for(int i = 0; i < k; i++) {
            currSum += nums[i];   
        }
        
        currAverage = currSum/k;
        double ans = currAverage;
        for (int j = k; j < nums.length; j++) {
            currSum  = currSum + nums[j] - nums[j - k];
            currAverage = currSum/k;
            ans = Math.max(ans, currAverage);
        }
        
        return ans;
    }
}

File: leetcode-submissions/0643-maximum-average-subarray-i/solution.py
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        current = 0
        
        i = 0
        while i in range(k):
            current += nums[i]
            i += 1
            
        average = current / k
        
        left = 0
        right = k
        
        ans = average
        
        while right < len(nums):
            current -= nums[left]
            current += nums[right]
            
            average = current / k
            
            left += 1
            right += 1
            
            ans = max(ans, average)
            
        return ans
            
            
